plugins/content/caption/base/batch/ParseMultiLanguageCaptionAsset/KAsyncParseMultiLanguageCaptionAsset.class.php

```diff
@@ -120,7 +139,22 @@ protected function parseMultiLanguage(KalturaBatchJob $job, KalturaParseMultiLan
 		self::deleteCaptions($captionChildernIds);
 		self::unimpersonate();
 
-		$this->closeJob($job, null, null, "Finished parsing", KalturaBatchJobStatus::FINISHED);
+		$errNumber = null;
+		if ($captionsCreated)
+		{
+			$errType = null;
+			$message = CaptionPlugin::FINISHED_PARSING_MESSAGE;
+			$status = KalturaBatchJobStatus::FINISHED;
+			$data = null;
+		}
+		else
+		{
+			$errType = KalturaBatchJobErrorTypes::KALTURA_API;
```

Comment:
I agree with the KalturaBatchJobErrorTypes::APP.
closeJobOnError requires an Exception as a third argument.
I'm closing the job explicitly without an error so i'm using the closeJob function.